src/adapter/src/catalog/migrate.rs

```diff
@@ -283,10 +282,15 @@ fn ast_rewrite_create_sink_into_kafka_options_0_80_0(
         fn visit_create_sink_statement_mut(&mut self, node: &'ast mut CreateSinkStatement<Raw>) {
             match &mut node.connection {
                 CreateSinkConnection::Kafka { options, .. } => {
-                    options.push(KafkaSinkConfigOption {
-                        name: KafkaSinkConfigOptionName::LegacyIds,
-                        value: Some(WithOptionValue::Value(Value::Boolean(true))),
-                    });
+                    if !options
+                        .iter()
+                        .any(|option| matches!(option.name, KafkaSinkConfigOptionName::LegacyIds))
+                    {
+                        options.push(KafkaSinkConfigOption {
+                            name: KafkaSinkConfigOptionName::LegacyIds,
+                            value: Some(WithOptionValue::Value(Value::Boolean(true))),
+                        });
+                    }
```

Comment:
A bit nervous about this, because the version conditional on calling this function _should_ have ensured this function was only ever called once. It's a bit disconcerting if you were seeing this function called multiple times, because that might indicate this was erroneously getting invoked _post_ v0.80, and injecting `LEGACY IDS = TRUE` where it shouldn't've. 

I'd feel better if you just totally nuked all these migrations. Since we're on v0.86 now I don't think we need literally any of the migrations that are currently in this file?